src/types/CustomType.ts

```diff
@@ -0,0 +1,9 @@
+import type { BscType } from './BscType';
+import { ObjectType } from './ObjectType';
+
+export class CustomType extends ObjectType implements BscType {
+
+    constructor(public customTypeName: string) {
```

Comment:
No need for `customType` to be in the `customTypeName` property name. Just call it `name`. Similarto how the `FunctionType` has a `name` property. https://github.com/rokucommunity/brighterscript/blob/b69b86259fed32b69841d87c04ca107deb2f3dbc/src/types/FunctionType.ts#L14